### ISSUE TITLE
share HTML presentation, not just PDF

### ISSUE THREAD

**Issue by nchauvat** [bigbluebutton/bigbluebutton#9235]
**Is your feature request related to a problem? Please describe.**

I commonly make my presentations using https://revealjs.com/ and would have liked not to have to convert them to pdf to share them in BBB.

**Describe the solution you'd like**

When I click on the button "share presentation", I could pick the index.html of my presentation, or a zip that contains the html and the related static files.

The presentation would be shared in the room as it is currently for pdf presentations.

**Describe alternatives you've considered**

I currently share my screen and display my HTML presentation in my browser.

By doing this, I do not have access to the functionnalities that are available for presentations, like drawing, polling, etc.

**Additional context**

Thank you for the work done on Big Blue Button !

**Comment by mikhailnov** [comment]
Can LibreOffice convert your HTML to PDF?

**Comment by nchauvat** [comment]
I just gave it a try, but it looks like there is too much CSS involved for LibreOffice to cope. Unfortunately, the answer is no.

You may also want to know that presentations made with revealjs can be two-dimensional. Moving horizontaly you change chapter/section and moving verticaly you move to the next page in the chapter/section.

I do not know of a way to do the same with a PDF, for PDF are a linear sequence of pages.

**Comment by nchauvat** [comment]
Please note that my question was about using a dynamic HTML document as a presentation format. If the only possible format for presentation is PDF, I am left with converting a dynamic HTML into a static PDF or sharing my screen to display my own browser.

But when I share my screen, I lose the ability to display polls or draw with the mouse. If I could use polls when sharing screen and allow participants to draw with my shared screen in the background, I would have my use case covered.

**Comment by ostasevych** [comment]
I see that you are about iframing some linked HTML files, so is this about cms features?

**Comment by nchauvat** [comment]
@ostasevych I am sorry, I do not understand your question. I tried to describe a use case and I am not concerned about the technical solution. If it can be solved with an iframe, so be it. I do not know what are the "cms features" you mention. Could you explain ?

**Comment by arthurlogilab** [comment]
Here is an example you can use , showing the following presentations in bbb : 

* http://slides.logilab.fr/pyconfr2015_import_donnees_cubicweb/
* https://hackmd.logilab.fr/p/B1xHsbuT7#/ 

I don't think this is a CMS feature, we don't want to edit content. Just share an HTML presentation without converting to PDF (which depending on the framework used can be cumbersome) 

Technically (without further knowledge of bbb) I can see at least two ways forward : 

* the import PDF document which converts to images could be adapted to use something like https://wkhtmltopdf.org/ (rendering may vary), other tools may exist
* an iframe could be used with some sort of javascript magic to syncronize the anchor url accross all participants (like https://togetherjs.com/ or https://socket.io/)

**Comment by ptandler** [comment]
This might be related to #9785 (share website)

**Comment by hex-m** [comment]
LibreOffice Impress (and seemingly also [PowerPoint](https://support.microsoft.com/en-us/office/publish-your-presentation-to-the-web-558ae647-9919-4571-86a0-9a4a1ea48938)) can save/export a presentation as HTML. Supporting this would make it possible to have [clickable links](https://github.com/bigbluebutton/bigbluebutton/issues/6708) and other interactive content in a presentation.

somewhat off-topic: In the current conversion workflow (pptx/odp/.. -> pdf -> svg) a lot of details gets lost. Except for using HTML it might also be possible to directly render PDF or OpenDocument (with ViewerJS) in the Browser.

**Comment by tbussmann** [comment]
@stalebot: please keep this. 

Sharing HTML would still be useful. I'm just not sure if this can be done in a save (sandboxed) way and how to transfer the interaction with the presentation (scroll position, next slide commands) to the other users in a generic way. But like the Youtube sharing functionality, there could be a specific support for reveal.js and the like.

**Comment by nidico** [comment]
@stale: unstale!

**Comment by SebastianGrau** [comment]
I would be very much interested in the possibility to use reveal.js slides instead of PDF only.

**Comment by dbwelch** [comment]
Yes, I would like dynamic content in the slides, which can be served easily via html/url.  Is there another method to do this other than through a presentation?  It could via the "share video", but, this is only via specific video services, could this be adapted easily if presentation is not?

**Comment by hex-m** [comment]
It is planned to enable BBB to embed HTML-content. -> https://github.com/bigbluebutton/bigbluebutton/issues/15482

**Comment by ptandler** [comment]
any update?

**Comment by antobinary** [comment]
Perhaps this would be of use: https://github.com/bigbluebutton/plugin-generic-link-share
